src/sys/socket/sockopt.rs

```diff
@@ -478,6 +485,67 @@ unsafe impl<'a> Set<'a, usize> for SetUsize {
     }
 }
 
+/// Size of empty CString buffer
+/// There is no default size in kernel documentation
+const SIZE_CSTRING_MAX : usize = 64;
+
+/// Getter for a `CString` value.
+struct GetCString {
+    len: socklen_t,
+    val: [u8; SIZE_CSTRING_MAX],
+}
+
+unsafe impl Get<CString> for GetCString {
+    unsafe fn blank() -> Self {
+        GetCString {
+            len: SIZE_CSTRING_MAX as socklen_t,
+            val: [0u8; SIZE_CSTRING_MAX],
+        }
+    }
+
+    fn ffi_ptr(&mut self) -> *mut c_void {
+        self.val.as_mut_ptr() as *mut c_void
+    }
+
+    fn ffi_len(&mut self) -> *mut socklen_t {
+        &mut self.len
+    }
+
+    unsafe fn unwrap(self) -> CString {
+        // Crop length of val with the first null byte (because CString::new add null byte)
+        let len = {
+            let mut i = 0;
+            loop {
+                if i == self.len || self.val[i as usize] == 0 {
```

Comment:
You can replace the entire body of this method with `CStr::from_ptr(self.ffi_ptr()).to_owned()`.  Or, if you replace `CString` with `OsString`, you can use `OsStr::from_bytes(self.val).to_owned()`.